src/shared/components/ui/search-highlight.tsx

```
@@ -0,0 +1,147 @@
+    }
+    
+    // Add remaining text
+    result += trimmedText.slice(lastIndex);
+    
+    return result;
+  }, [text, searchTerm, highlightClassName, maxLength]);
+  
+  return (
+    <span 
+      className={className}
+      dangerouslySetInnerHTML={{ __html: highlightedText }}
```

Biome (2.1.2)

[error] Avoid passing content using the dangerouslySetInnerHTML prop.

Setting content using code can expose users to cross-site scripting (XSS) attacks

(lint/security/noDangerouslySetInnerHtml)